test/utils.c

```diff
@@ -92,7 +104,7 @@ void start_meshlink_pair(meshlink_handle_t *a, meshlink_handle_t *b) {
 	meshlink_start(a);
 	meshlink_start(b);
 
-	assert(wait_sync_flag(&pair_status, 5));
+	assert(wait_sync_flag_ex(&pair_status, 5));
```

Comment:
Same here, we should keep using `wait_sync_flag()` here, but remove the calls to `pthread_mutex_lock/unlock()` around it.